src/editor/components/groupBy/QueryEditorGroupBy.tsx

```diff
@@ -47,9 +49,10 @@ export const QueryEditorGroupBy: React.FC<Props> = props => {
         interval,
       };
 
-      props.onChange(payload);
+      onChange(payload); // adding onChange to dependency array below causes maximum call depth error
```

Comment:
I wrote a comment about this one. If I add onChange to dependencies we get a recursion with a maximum call stack depth. Didn't look into this further, but we really should dive deeper into this when we have time.

Reply:
hmm, it looks like we are doing something really bad here and it should probably be changed. Lets add this one to the list. I'm not really following what we are trying to solve so need to have a look at that before I can suggest a solution.